File: frontend/src/types/scheduledReport.ts
/**
 * @file types/scheduledReport.ts
 * @description 定时报表类型定义
 */

export interface ScheduledReport {
  id: number;
  savedReportId: number | null;
  cronExpr: string;
  recipientEmails: string;
  format: 'PDF' | 'EXCEL';
  status: 'ACTIVE' | 'PAUSED';
  lastRunAt: string | null;
  nextRunAt: string | null;
  subject: string;
  createTime: string;
}

File: backend/src/main/java/com/ams/entity/ScheduledReport.java
package com.ams.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("scheduled_report")
public class ScheduledReport {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long savedReportId;
    private String cronExpr;
    private String recipientEmails;
    private String format;
    private String status;
    private LocalDateTime lastRunAt;
    private LocalDateTime nextRunAt;
    private String subject;

    private String tenantId;
    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}
